client/controller/web-util.js

```diff
@@ -1,9 +1,10 @@
-import ReactDom from 'react-dom';
+import { hydrateRoot } from 'react-dom/client';
+import { getRootNode, getRootDomElement } from 'calypso/boot/common';
 
 export function render( context ) {
-	ReactDom.render( context.layout, document.getElementById( 'wpcom' ) );
+	getRootNode().render( context.layout );
 }
 
 export function hydrate( context ) {
-	ReactDom.hydrate( context.layout, document.getElementById( 'wpcom' ) );
+	hydrateRoot( getRootDomElement(), context.layout );
```

Comment:
This is a mistake. `hydrateRoot` returns the React root object that you should store and use for subsequent renders. But instead you're throwing the root away, and try to create a new one with `createRoot`, on the same element.

Reply:
Thanks for the great catch. I've added a way to store the hydrated root here: https://github.com/Automattic/wp-calypso/pull/95513/commits/c6ec090ab4b89b6ac48b6bf28fc3e6ed82198aac . If that works for you, I'd want to enhance these minimal root node managing APIs though. Currently there is nothing preventing anyone from calling `setRootNode()` in some random places, which I'm not sure if it'd make sense to calypso. 
For example, a singleton like thing that can only be initialized once across the app, either by passing no argument like `render()` or by passing an prepared `Root` like `hydrate()`. After that, the stored `Root` would remain unchanged.